lib/hyrax/transactions/steps/delete_access_control.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+require 'dry/monads'
+
+module Hyrax
+  module Transactions
+    module Steps
+      ##
+      # Deletes the Hyrax::AccessControlList for any resource with a `#permission_manager`.
+      # If `#permission_manager` is undefined, succeeds.
+      #
+      # @see https://dry-rb.org/gems/dry-monads/1.0/result/
+      class DeleteAccessControl
```

Comment:
this looks great.

i'm curious whether this logic wouldn't be more effective as a listener on `object/collection.deleted`? the considerations i have in mind are:
  - are there common cases where you'd want to delete an object or collection using the persister, but NOT cleanup this access control? a listener is closer to universal than a transaction step, so this could be a good reason to switch.
  - if the destroy step succeeds, but the object acl deletion fails, will we want to report an error back to the controller? if so, i'd think that's a point in favor of the transaction step; but maybe we'd rather log and continue processing, leaving the ACL orphaned?

i don't want to hold this up on the answer either way. it's easy enough to get this in and rework the implementation if we decide a listener is better.

Reply:
Good point.  I was following the pattern of the save_access_control step.  So I wonder if it should happen the same way too.

Reply:
But I guess if the save ACL step fails then we would want to provide feedback to the user in the controller.

Reply:
my first intuition is that save is different on both counts. for save:
  - yes, we do commonly want to save only the metadata and not the access controls---the need to check access control save is contextual to the transaction.
  - yes, we do want to fail visibly if we've saved an object but not been able to apply requested ACLs.